preprocessing.py

```diff
@@ -78,11 +83,28 @@ def sliding_window(size_of_window, docid, dict_of_all_terms):
     for current_window in range(0, k):
         current_vector = sparse.lil_matrix((dimension, 1))
         for e in range(0, size_of_window):
-            current_vector[(dict_of_all_terms[data[e + size_of_window * current_window ]], 0)] += 1
+            current_vector[(dict_of_all_terms[data[e + size_of_window *
+                                                   current_window]], 0)] += 1
         result.append(current_vector)
+    projector_contruction(result)
     return result
 
 
+def projector_contruction(dict_of_vectors, weight_distribution=None):
+    dimension = dict_of_vectors[0].shape[0]
+    s = sparse.lil_matrix((dimension, dimension))
+    k = 0
+    if weight_distribution is not None:
+        for item in dict_of_vectors:
```

Comment:
for (vec, weight) in zip(dict_of_vectors, weight_distribution):
   s += weight * vec.dot(vec.transpose())

such code more pretty and "k" variable is useless

Reply:
moreover,
you can write code like:

import itertools as it

weights = it.repeat(1.0 / len(dict_of_vectors)) if weight_distribution is None else weight_distribution
for (vec, weight) in zip(dict_of_vectors, weights):
--- s += weight * vec.dot(vec.transpose())